### ISSUE TITLE
[FEATURE_REQUEST]  Make the "no pods found matching service labels" error more specific

### ISSUE THREAD

**Issue by marcindulak** [stackrox/kube-linter#249]
**Description of the problem/feature request**

If possible, a more targeted error which provides information about the source of the problem is reported by kube-linter.

The "no pods found matching service label" error seems to trigger when certain problems are present in the pod or container spec.
In the example below `dnsPolicy: {{ }}` is used to trigger the error message, but can also be triggered by using `name: {{ }}` for the container name, and other similar ways.

**Description of the existing behavior vs. expected behavior**

```
kube-linter lint - <<EOF --do-not-auto-add-defaults --include dangling-service
apiVersion: apps/v1
kind: Deployment
metadata:
  name: test
spec:
  selector:
    matchLabels:
      app: test
  template:
    metadata:
      labels:
        app: test
    spec:
      containers:
      - image: nginx:latest
        name: test
      dnsPolicy: {{ }}
---
apiVersion: v1
kind: Service
metadata:
  name: test
  labels:
    app: test
spec:
  ports:
  - name: http
    port: 8080
    targetPort: 8080
  selector:
    app: test
  type: ClusterIP
EOF
```

Output
```
KubeLinter 0.2.5

<standard input>: (object: <no namespace>/test /v1, Kind=Service) no pods found matching service labels (map[app:test]) (check: dangling-service, remediation: Confirm that your service's selector correctly matches the labels on one of your deployments.)

Error: found 1 lint errors
```

**Additional context**

```
cat /etc/*release | grep PRETTY
PRETTY_NAME="Ubuntu 20.04.3 LTS"
```

```
kube-linter version
0.2.5
```

Downloaded as tar.gz from https://github.com/stackrox/kube-linter/releases/tag/0.2.5

**Comment by marcindulak** [comment]
I'm thinking about getting some lower level information about the reason for the pod being skipped due to an incorrect spec

https://github.com/stackrox/kube-linter/blob/b614a5d808aaead38a4b92553a2f3fb738ec28db/pkg/templates/danglingservice/template.go#L51-L55

The `PodTemplateSpec` may be aware of more details why the pod is not valid, but it does not propagate this information higher up

https://github.com/stackrox/kube-linter/blob/b614a5d808aaead38a4b92553a2f3fb738ec28db/pkg/extract/pod_spec.go#L13

In this specific example, when `-v` option is added as hinted in https://github.com/stackrox/kube-linter/issues/63, a parse "Warning" is printed (to stderr), but it's not clear where does it originate from. Maybe it could be part of the stdout error output somehow?

```
kube-linter lint - <<EOF --do-not-auto-add-defaults --include dangling-service -v
...
Warning: failed to load object from <standard input>: failed to decode: yaml: invalid map key: map[interface {}]interface {}{}
KubeLinter 0.2.5

<standard input>: (object: <no namespace>/test /v1, Kind=Service) no pods found matching service labels (map[app:test]) (check: dangling-service, remediation: Confirm that your service's selector correctly matches the labels on one of your deployments.)

Error: found 1 lint errors
```

In general, a schema validation would capture other types of spec errors https://github.com/stackrox/kube-linter/issues/36

**Comment by viswajithiii** [comment]
Yeah, this is a valid problem, and should be solved by fixing #63. @janisz This is probably something we should prioritize, because there are many problems that manifest from this root cause.

**Comment by janisz** [comment]
@marcindulak Does solution from #279 solves this? With new flag it will looks like this:

```yaml
./kube-linter lint - <<EOF --do-not-auto-add-defaults --include dangling-service --fail-on-invalid-resource
apiVersion: apps/v1
kind: Deployment
metadata:
  name: test
spec:
  selector:
    matchLabels:
      app: test
  template:
    metadata:
      labels:
        app: test
    spec:
      containers:
      - image: nginx:latest
        name: test
      dnsPolicy: {{ }}
---
apiVersion: v1
kind: Service
metadata:
  name: test
  labels:
    app: test
spec:
  ports:
  - name: http
    port: 8080
    targetPort: 8080
  selector:
    app: test
  type: ClusterIP
EOF
KubeLinter development

<standard input>: (object: <no namespace>/test /v1, Kind=Service) no pods found matching service labels (map[app:test]) (check: dangling-service, remediation: Confirm that your service's selector correctly matches the labels on one of your deployments.)

<standard input>: (object: <no namespace>/ /, Kind=) failed to decode: yaml: invalid map key: map[interface {}]interface {}{} (check: failed-to-load-object, remediation: Confirm that the file is accessible and is valid k8s yaml.)
```
